src/Cake.Common/Build/TeamCity/Data/TeamCityPullRequestInfo.cs

```diff
@@ -11,6 +11,39 @@ namespace Cake.Common.Build.TeamCity.Data
     /// </summary>
     public class TeamCityPullRequestInfo : TeamCityInfo
     {
+        private static bool InferIsPullRequest(string gitReferenceName)
+        {
+            var branchSlices = gitReferenceName.Split('/');
+
+            if (branchSlices.Length >= 3)
+            {
+                switch (branchSlices[1].ToUpper())
+                {
+                    case "CHANGES":
+                    case "MERGE-REQUESTS":
+                    case "PULL":
+                    case "PULL-REQUESTS":
+                        return true;
+                    default:
+                        return false;
+                }
+            }
+
+            return false;
+        }
+
+        private static int? GetPullRequestNumber(string gitReferenceName)
+        {
```

Comment:
Should check & return if `gitReferenceName` is null or empty.

Reply:
done